src/app.py

```diff
@@ -49,18 +52,15 @@ def is_anonymous(self) -> bool:
 
     @property
     def name(self) -> str:
-        if self.is_anonymous:
+        if self.__token is None:
             return "anonymous"
-        # This feature is still WIP. TODO - token parsing and validating
-        return "username"
+        return self.__token.get("preferred_username", "unknown")
 
-    @property
-    def roles(self) -> List[str]:
-        if self.is_anonymous:
-            # Temporarily enable everything to everyone
-            return ["loggedin", "admin"]
-        # This feature is still WIP. TODO - user roles and configuration
-        return ["loggedin", "admin"]
+    def roles(self, client_id: Optional[str]) -> List[str]:
+        if self.__token is None:
+            return []
+        access = self.__token.get("resource_access", {})
+        return access.get(client_id, {}).get("roles", [])
```

Comment:
not a :fan: of this - maybe just split it into one or two if-cases?

Reply:
Not sure. I'll experiment with unrolling it a bit (there are as much as three `get`s in this method)

Reply:
Consider this: https://devblogs.microsoft.com/python/idiomatic-python-eafp-versus-lbyl/